src/python/pants/goal/anonymous_telemetry.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import json
+import logging
+import re
+import uuid
+from typing import cast
+
+from humbug.consent import HumbugConsent  # type: ignore
+from humbug.report import Modes, Report, Reporter  # type: ignore
+
+from pants.engine.internals.scheduler import Workunit
+from pants.engine.rules import collect_rules, rule
+from pants.engine.streaming_workunit_handler import (
+    StreamingWorkunitContext,
+    WorkunitsCallback,
+    WorkunitsCallbackFactory,
+    WorkunitsCallbackFactoryRequest,
+)
+from pants.engine.unions import UnionRule
+from pants.option.subsystem import Subsystem
+from pants.util.docutil import docs_url
+
+logger = logging.getLogger(__name__)
+
+
+_bugout_access_token = "3ae76900-9a68-4a87-a127-7c9f179d7272"
+_bugout_journal_id = "801e9b3c-6b03-40a7-870f-5b25d326da66"
+_telemetry_docs_url = docs_url("anonymous-telemetry")
+_telemetry_docs_referral = f"See {_telemetry_docs_url} for details"
+
+
+class AnonymousTelemetry(Subsystem):
+    options_scope = "anonymous-telemetry"
+    help = "Options related to sending anonymous stats to the Pants project, to aid development."
+
+    @classmethod
+    def register_options(cls, register):
+        register(
+            "--enabled",
+            advanced=True,
+            type=bool,
+            default=False,
+            help=(
+                f"Whether to send anonymous telemetry to the Pants project.\nTelemetry is sent "
+                f"asynchronously, with silent failure, and does not impact build times or "
+                f"outcomes.\n{_telemetry_docs_referral}."
+            ),
+        )
+        register(
+            "--repo-id",
+            advanced=True,
+            type=str,
+            default=None,
+            help=(
+                f"An anonymized ID representing this repo.\nFor private repos, you likely want the "
+                f"ID to not be derived from, or algorithmically convertible to, anything "
+                f"identifying the repo.\nFor public repos the ID may be visible in that repo's "
+                f"config file, so anonymity of the repo is not guaranteed (although user anonymity "
+                f"is always guaranteed).\n{_telemetry_docs_referral}."
+            ),
+        )
+
+    @property
+    def enabled(self) -> bool:
+        return cast(bool, self.options.enabled)
+
+    @property
+    def repo_id(self) -> str | None:
+        return cast("str | None", self.options.repo_id)
+
+
+class AnonymousTelemetryCallback(WorkunitsCallback):
+    def __init__(self, anonymous_telemetry: AnonymousTelemetry) -> None:
+        super().__init__()
+        self._anonymous_telemetry = anonymous_telemetry
+
+    @property
+    def can_finish_async(self) -> bool:
+        # Because we don't log anything, it's safe to finish in the background.
+        return True
```

Comment:
Maybe mention this in the docs, that the report runs in the background after your Pants run to avoid slowing you down? I could imagine users being concerned if every run will now make additional network requests.

Reply:
Done.